extensions/OES_EGL_image_external/extension.xml

```diff
@@ -0,0 +1,169 @@
+        <stage type="fragment"/>
+        <type name="samplerExternalOES"/>
+
+        <function name="texture2D" type="vec4">
+          <param name="sampler" type="samplerExternalOES"/>
+
+          <param name="coord" type="vec2"/>
+        </function>
+
+      </glsl>
+    </features>
+  </overview>
+
+  <idl xml:space="preserve">
+[NoInterfaceObject]
+interface OESEGLImageExternal {
+    const GLenum TEXTURE_EXTERNAL_OES             = 0x8D65;
+    const GLenum SAMPLER_EXTERNAL_OES             = 0x8D66;
+    const GLenum TEXTURE_BINDING_EXTERNAL_OES     = 0x8D67;
+    const GLenum REQUIRED_TEXTURE_IMAGE_UNITS_OES = 0x8D68;
+
+    [RaisesException] void EGLImageTargetTexture2DOES(
+        GLenum target, HTMLVideoElement video);
+};
+  </idl>
+
+  <!-- New Implementation-Dependent State -->
+  <samplecode xml:space="preserve">
+
+    <p> This a fragment shader that samples a video texture.</p>
+    <pre>
+    #extension GL_OES_EGL_image_external : require
+    precision mediump float;
+    varying vec2 v_texCoord;
+
+    uniform samplerExternalOES uSampler;
+
+    void main(void) {
+      gl_FragColor = texture2D(uSampler, v_texCoord);
+    }
+    </pre>
+
+    <p> This shows application that renders video using proposed extension.  </p>
+    <pre>
+    var videoElement = document.getElementById("video");
+    var videoTexture = gl.createTexture();
+
+    function update() {
+        var ext = gl.getExtension('OES_EGL_image_external');
+        if(ext !=== null){
+            gl.bindTexture(ext.TEXTURE_EXTERNAL_OES, videoTexture);
+            ext.EGLImageTargetTexture2DOES(ext.TEXTURE_EXTERNAL_OES, videoElement);
+            gl.bindTexture(ext.TEXTURE_EXTERNAL_OES, null);
+        }
+    }
+
+    function render() {
+        gl.clearColor(0.0, 0.0, 1.0, 1.0);
+        gl.clear(gl.COLOR_BUFFER_BIT);
+
+        gl.bindBuffer(gl.ARRAY_BUFFER, squareVerticesBuffer);
+        gl.vertexAttribPointer(vertexPositionAttribute, 3, gl.FLOAT, false, 0, 0);
+
+        gl.activeTexture(gl.TEXTURE0);
+        gl.bindTexture(ext.TEXTURE_EXTERNAL_OES, videoTexture);
+        gl.uniform1i(gl.getUniformLocation(shaderProgram, "uSampler"), 0);
+
+        gl.drawArrays(gl.TRIANGLE_STRIP, 0, 4);
+    }
+    </pre>
+
+    <p> Application renders each video frames into WebGL canvas based on game-loop pattern. </p>
+    <pre>
+    update();
+
+    while (true) {
+       processInput();
+       render();
+    }
+    </pre>
+
+  </samplecode>
+
+  <tests/>
+
+  <issues/>
+
+  <history>
+    <revision date="2016/10/11">
+      <change>Initial revision.</change>
+    </revision>
+    <revision date="2016/10/11">
+      <change>Refine sample code.</change>
+    </revision>
+    <revision date="2016/10/11">
+      <change>Remove comment on WebGL API 2.0 dependency</change>
+      <change>Refine sample code as an implicit texture updating mode</change>
+    </revision>
+    <revision date="2016/11/01">
+      <change>Remove comment.</change>
+    </revision>
```

Comment:
This is the first commit of this extension, so please feel free to collapse these history items if you see fit.